test/distributed/org/apache/cassandra/distributed/test/streaming/BoundExceptionTest.java

```diff
@@ -77,19 +73,16 @@ public void testExceptionCycle()
         e1.initCause(e2);
         e2.initCause(e1);
 
-        Throwable boundedStackTrace = StreamSession.boundStackTrace(e1, LIMIT);
-        String expectedStackTrace1 = "java.lang.Exception: Test exception 1\n" +
-                                    "\tat org.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testExceptionCycle(BoundExceptionTest.java:74)\n";
-        String expectedStackTrace2 = "java.lang.RuntimeException: Test exception 2\n" +
-                                     "\tat org.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testExceptionCycle(BoundExceptionTest.java:75)\n";
-        String expectedStackTrace3 = "[CIRCULAR REFERENCE: java.lang.Exception: Test exception 1]\n";
-
-        String boundedStackTraceAsString = Throwables.getStackTraceAsString(boundedStackTrace);
+        StringBuilder boundedStackTrace = StreamSession.boundStackTrace(e1, LIMIT);
+        String expectedStackTrace = "java.lang.Exception: Test exception 1\n" +
+                                     "\torg.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testExceptionCycle(BoundExceptionTest.java:70)\n" +
+                                     "\tjava.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n" +
+                                     "java.lang.RuntimeException: Test exception 2\n" +
+                                     "\torg.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testExceptionCycle(BoundExceptionTest.java:71)\n" +
+                                     "\tjava.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n" +
+                                     "[CIRCULAR REFERENCE: java.lang.Exception: Test exception 1]";
 
-        assertTrue(boundedStackTraceAsString.contains(expectedStackTrace1));
-        assertTrue(boundedStackTraceAsString.contains(expectedStackTrace2));
-        assertTrue(boundedStackTraceAsString.contains(expectedStackTrace3));
-        assertEquals(boundedStackTrace.getStackTrace().length, LIMIT);
+        assertEquals(expectedStackTrace, boundedStackTrace.toString());
     }
 
     @Test
```

Comment:
you are missing a test where stack element length < limit, you have a bug in this case